sklearn/metrics/_regression.py

```diff
@@ -284,7 +284,7 @@ def mean_pinball_loss(
 
 
 def mean_absolute_percentage_error(
-    y_true, y_pred, sample_weight=None, multioutput="uniform_average"
+    y_true, y_pred, *, sample_weight=None, multioutput="uniform_average"
```

Comment:
To be backward compatible, we can use `_deprecate_positional_args`:

https://github.com/scikit-learn/scikit-learn/blob/48e83df4509bb2ae1c1d74570a9efed7b61bdede/sklearn/utils/validation.py#L36

to deprecate position arguments. Then a test needs to be added to make sure that the deprecation message is shown.

Reply:
I was thinking that we could consider it as a fix. But is true that the keyword-only will be enforced only from 1.1 so this might be better to make a deprecation cycle.